pkg/loadbalancer/server.go

```diff
@@ -65,10 +66,17 @@ func (s *Server) GetLoadBalancer(ctx context.Context, clusterName string, servic
 		}
 	}
 	if ipv4 != "" && svcIPv4 {
```

Comment:
This should do something like this, get the kubernetes version of the cluster (not call it each time)

```
sv, _ := kubeClient.ServerVersion()
```

and then add or skip the IPMode if is not supported as we do in kind

https://github.com/kubernetes-sigs/kind/blob/19df3db3ac3d717856f2e93537aa42b09a903706/pkg/cluster/internal/kubeadm/config.go#L477-L482

Reply:
Unless we're actually going to error out if `IPMode` is not supported (which would technically be correct since the only configuration we support is `Proxy` mode with `IP` set, which was not an allowed combination pre-`IPMode`), there is no real reason to care whether the apiserver or the service proxy knows about the existence of the `IPMode` field; the result of not setting the field is exactly the same as the result of setting it and having it be ignored.

Reply:
Will be it ignored?
If the API server receives an object with unknown field does it discard them?

I always forget about this, can we validate this assumption?

Reply:
if I try to use `kubectl replace --subresource status service ..` with a field that does not exist I got this error

```
I0505 09:44:01.157687 3769194 request.go:1212] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Service in version \"v1\" cannot be handled as a Service: strict decoding error: unknown field \"status.loadBalancer.ingress[0].nonvalid\"","reason":"BadRequest","code":400}
I0505 09:44:01.159088 3769194 helpers.go:246] server response object: [{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "error when replacing \"apache.yaml\": Service in version \"v1\" cannot be handled as a Service: strict decoding error: unknown field \"status.loadBalancer.ingress[0].nonvalid\"",
  "reason": "BadRequest",
  "code": 400
}]
Error from server (BadRequest): error when replacing "apache.yaml": Service in version "v1" cannot be handled as a Service: strict decoding error: unknown field "status.loadBalancer.ingress[0].nonvalid"

```